### PR TITLE
[OF-1086] test: Add tests for APIResponseBase

### PR DESCRIPTION
**Pull request by MAG-AW**
connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Tests/src/PublicAPITests/SystemResultTests.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+{
+public:
+	ResponseReceiver() : ResponseReceived(false), ThreadId(std::this_thread::get_id())
+	{
+	}
+
+	void OnHttpResponse(csp::web::HttpResponse& InResponse) override
+	{
+		// We expect the callback to have come from a seperate Thread
+		// EXPECT_FALSE(std::this_thread::get_id() == ThreadId);
```
Comment (MAG-SamBirley) [inline Tests/src/PublicAPITests/SystemResultTests.cpp:81]: Can we either remove this commented-out line, or (even better) uncomment it and validate the assertion?

File: Tests/src/PublicAPITests/SystemResultTests.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+private:
+	csp::web::HttpResponse Response;
+
+	std::atomic<bool> ResponseReceived;
+	std::thread::id ThreadId;
+};
+
+void NullResultTestFunction(NullResultCallback Callback)
+{
+	csp::systems::NullResult InternalResult(csp::systems::EResultCode::Success, (uint16_t) csp::web::EResponseCodes::ResponseNoContent);
```
Comment (MAG-SamBirley) [inline Tests/src/PublicAPITests/SystemResultTests.cpp:104]: Could we test that this does not throw an exception with a try/catch block?

File: Tests/src/PublicAPITests/SystemResultTests.cpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	const csp::common::String MyTestPayload			  = "1234";
+
+	auto* WebClient = CSP_NEW TestWebClient(80, csp::web::ETransferProtocol::HTTP);
+	EXPECT_TRUE(WebClient != nullptr);
+
+	ResponseReceiver Receiver;
+
+	// Synthesise a response to feed to ApiResponseBase
+	csp::web::HttpRequest MyTestRequest = csp::web::HttpRequest(WebClient,
+																csp::web::ERequestVerb::GET,
```
Comment (MAG-SamBirley) [inline Tests/src/PublicAPITests/SystemResultTests.cpp:133]: Could we address the whitespacing here?

**Review by MAG-SamBirley — CHANGES_REQUESTED** [review]

File: Tests/src/PublicAPITests/SystemResultTests.cpp (hunk 3 above)
Comment (MAG-SamBirley) [reply Tests/src/PublicAPITests/SystemResultTests.cpp:138]: It looks the same? I guess its the auto-clang-format?

Comment (MAG-SamBirley) [reply Tests/src/PublicAPITests/SystemResultTests.cpp:138]: Ok. no worries then. Thanks!

**Review by MAG-SamBirley — APPROVED** [review]

**Review by MAG-AdamThorn — APPROVED** [review]

**Pull request merged**